### PR TITLE
Use Metal on iOS by default.

### PR DESCRIPTION
**Pull request by chinmaygarde**
Fixes https://github.com/flutter/flutter/issues/18208. All Metal for iOS related work items described in https://github.com/orgs/flutter/projects/5 have been completed.

### CODE REVIEW

File: shell/platform/darwin/ios/rendering_api_selection.mm [hunk 1]
```diff
@@ -15,40 +15,17 @@
 
 namespace flutter {
 
-bool ShouldUseSoftwareRenderer() {
-  return [[[NSProcessInfo processInfo] arguments] containsObject:@"--force-software"];
-}
-
```
Comment (dnfield) [inline shell/platform/darwin/ios/rendering_api_selection.mm:21]: Why are we dropping this?

Comment (chinmaygarde) [reply shell/platform/darwin/ios/rendering_api_selection.mm:21]: This was only added recently so I could check all three variants in one build. There is no reason to support software rendering on iOS.

Comment (dnfield) [reply shell/platform/darwin/ios/rendering_api_selection.mm:21]: So it's set somewhere else for simulator?

Comment (chinmaygarde) [reply shell/platform/darwin/ios/rendering_api_selection.mm:21]: Roger https://github.com/flutter/engine/pull/17431/files#diff-68cfd589fe3ae373eb867fda891a0c40L56

**Pull request merged**